uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -825,6 +902,47 @@ define([
       }
 
       /**
+       * Loads the comments for current leave request
+       *
+       * @return {Promise}
+       */
+      function loadCommentsandContactnames() {
+        //In CREATE mode dont fetch comments
+        if(!this.isMode('create')) {
+          return this.request.loadComments()
+            .then(loadContactNames.bind(this));
+        }
+
+        return $q.resolve();
+      }
+
+      /**
+       * Loads unique contact names for all the comments
+       *
+       * @return {Promise}
+       */
+      function loadContactNames () {
+        var contactIDs = [],
+          self = this;
+
+        _.each(self.request.comments, function (comment) {
+          //Push only unique contactId's which are not same as logged in user
+          if(comment.contact_id != self.directiveOptions.contactId && contactIDs.indexOf(comment.contact_id) === -1) {
+            contactIDs.push(comment.contact_id);
+          }
+        });
+
+        return Contact.all({
+          id: { IN: contactIDs }
+        },{
+          page: 1,
+          size: 0
+        }).then(function (contacts) {
+          self.comments.contacts = _.indexBy(contacts.list, 'contact_id');
```

Comment:
why maintain a separate comments array when one is already available on leave request instance?